private/model/api/operation.go

```diff
@@ -366,7 +366,7 @@ func (c *{{ .API.StructName }}) {{ .ExportedName }}PagesWithContext(` +
 	}
 
 	cont := true
-	for p.Next() && cont {
+	for cont && p.Next() {
 		cont = fn(p.Page().({{ .OutputRef.GoType }}), !p.HasNextPage())
 	}
```

Comment:
I think if we modify this slightly we can remove the cognitive overhead of determining if `p.Next` will be called at all by updating the body body to use a `if` and `break` if the callback function returns `false`.

Getting rid of `cont` and doing the check inline would reduce the complexity here.

```go
	for p.Next() {
		if !fn(p.Page().({{ .OutputRef.GoType }}), !p.HasNextPage()) {
			break
		}
	}
```

Reply:
Sure.

Reply:
done in 098ea80c